app/src/main/java/org/sopt/and/presentation/signup/SignUpViewModel.kt

```diff
@@ -0,0 +1,85 @@
+package org.sopt.and.presentation.signup
+
+import androidx.lifecycle.ViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+import android.util.Patterns
+import java.util.regex.Pattern
+import org.sopt.and.domain.User
+
+data class SignUpUiState(
+    val email: String = "",
+    val password: String = "",
+    val showPassword: Boolean = false,
+    val isSignUpSuccess: Boolean = false,
+    val errorMessage: String? = null
+)
+
+sealed class SignUpEvent {
+    data object Success : SignUpEvent()
+    data class Error(val message: String) : SignUpEvent()
+}
+
+class SignUpViewModel : ViewModel() {
+    private val _uiState = MutableStateFlow(SignUpUiState())
+    val uiState: StateFlow<SignUpUiState> = _uiState.asStateFlow()
+
+    fun onEmailChange(email: String) {
+        _uiState.update { it.copy(
+            email = email,
+            errorMessage = null
+        ) }
```

Comment:
update를 사용하신다면 내부에서 it.copy를 사용하지 않으셔도 돼요.
update 함수 내부에서 변경점만 찾아서 변경하고, 기존 값들은 그대로 가져오는 로직이 구현되어있기 때문에 아래처럼 작성하셔도 동작할거에요
```suggestion
        _uiState.update {
            email = email,
            errorMessage = null
        }
```

Reply:
it.copy로 변경을 알려줘야 한다고 생각했는데 `.upadte`에 이해가 부족했네요ㅠ 제안 감사드립니다!

Reply:
헉 죄송합니다..!! 이부분은 제가 대충 보고 코리를 했는지 잘못 알려드렸네요 ㅜㅜ
지금 사용하신 방법이 맞습니다!

@angryPodo

Reply:
헉 다시 수정하겠습니다~